h/static/scripts/annotator/guest.coffee

```diff
@@ -287,40 +329,35 @@ module.exports = class Guest extends Annotator
 
   onFailedSelection: (event) ->
     @adder.hide()
-    @selectedTargets = []
+    @selectedRanges = []
 
-  # Select some annotations.
-  #
-  # toggle: should this toggle membership in an existing selection?
   selectAnnotations: (annotations, toggle) =>
+    this.triggerShowFrame()
     if toggle
-      # Tell sidebar to add these annotations to the sidebar if not already
-      # selected, otherwise remove them.
       this.toggleAnnotationSelection annotations
     else
-      # Tell sidebar to show the viewer for these annotations
-      this.triggerShowFrame()
       this.showAnnotations annotations
 
-  # When Mousing over a highlight, tell the sidebar to focus the relevant annotations
-  onAnchorMouseover: (event) ->
-    if @visibleHighlights
-      event.stopPropagation()
-      annotations = event.data.getAnnotations(event)
-      this.focusAnnotations annotations
-
-  # Tell the sidebar to stop highlighting the relevant annotations
-  onAnchorMouseout: (event) ->
-    if @visibleHighlights
-      event.stopPropagation()
-      this.focusAnnotations []
-
-  # When clicking on a highlight, tell the sidebar to bring up the viewer for the relevant annotations
-  onAnchorClick: (event) =>
-    if @visibleHighlights
-      event.stopPropagation()
-      this.selectAnnotations (event.data.getAnnotations event),
-        (event.metaKey or event.ctrlKey)
+  onHighlightMouseover: (event) ->
+    return unless @visibleHighlights
+    annotation = $(event.currentTarget).data('annotation')
+    annotations = event.annotations ?= []
+    annotations.push(annotation)
+    if event.target is event.currentTarget
+      setTimeout => this.focusAnnotations(annotations)
```

Comment:
These might need a comment. I'm letting these events bubble now and passing along an `Array` as `event.annotations`. The innermost handler fires off an asynchronous task here. By the time it runs, the `Array` is filled.

In some sense I find this cleaner than walking up the tree. It doesn't stop the event from bubbling, which is always better if possible (less interference with the page).